### PR TITLE
Runtime ISR binding and simple public API docs for RTC

### PR DESCRIPTION
**Pull request by JurajSadel**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description
Use runtime ISR binding for RTC and adds docs for public API.

#### Testing
Run modified clock_monitor and rtc_watchdog examples.

### CODE REVIEW

**Review by bjoernQ — APPROVED** [review]
LGTM - also added some more docs 👍

File: esp-hal/src/rtc_cntl/mod.rs [hunk 1]
```diff
@@ -201,14 +208,35 @@ impl<'d> Rtc<'d> {
         #[cfg(any(esp32, esp32s3, esp32c3, esp32c6))]
         RtcSleepConfig::base_settings(&this);
 
+        if let Some(interrupt) = interrupt {
+            unsafe {
+                crate::interrupt::bind_interrupt(
```
Comment (jessebraham) [inline esp-hal/src/rtc_cntl/mod.rs:213]: Can you please import the `crate::interrupt` module and the `crate::peripherals::Interrupt` type? There's no reason to include all these paths here, just makes it harder to read.

**Review by jessebraham — CHANGES_REQUESTED** [review]

File: esp-hal/src/rtc_cntl/mod.rs (hunk 1 above)
Comment (JurajSadel) [reply esp-hal/src/rtc_cntl/mod.rs]: Sure!

**Review by jessebraham — APPROVED** [review]
Thanks!

**Pull request merged**